src/TMxpProcessor.cpp

```diff
@@ -0,0 +1,161 @@
+ * * 2 - LOCKED LINE (until next newline) no MXP or HTML commands are
+ *     allowed in the line.  The line is not parsed for any tags at all.
+ *     This is useful for "verbatim" text output from the MUD.  When a
+ *     newline is received from the MUD, the mode reverts back to the
+ *     Default mode.
+ * The following additional modes were added to the v0.4 MXP spec:
+ * * 3 - RESET close all open tags.  Set mode to Open.  Set text color and
+ *     properties to default.
+ * * 4 - TEMP SECURE MODE set secure mode for the next tag only.  Must be
+ *     immediately followed by a < character to start a tag.  Remember to
+ *     set secure mode when closing the tag also.
+ * * 5 - LOCK OPEN MODE set open mode.  Mode remains in effect until
+ *     changed.  OPEN mode becomes the new default mode.
+ * * 6 - LOCK SECURE MODE set secure mode.  Mode remains in effect until
+ *     changed.  Secure mode becomes the new default mode.
+ * * 7 - LOCK LOCKED MODE set locked mode.  Mode remains in effect until
+ *     changed.  Locked mode becomes the new default mode."
+ */
+bool TMxpProcessor::setMode(int modeCode)
+{
+    // we really do not handle these well...
+    // MXP line modes - comments are from http://www.zuggsoft.com/zmud/mxp.htm#MXP%20Line%20TagsmMXP = true; // some servers don't negotiate, they assume!
+
+    mMXP = true;
+    switch (modeCode) {
+    case 0: // open line - only MXP commands in the "open" category are allowed.  When a newline is received from the MUD, the mode reverts back to the Default mode.  OPEN MODE starts as the Default mode until changes with one of the "lock mode" tags listed below.
+        mMXP_MODE = MXP_MODE_OPEN;
+        break;
+    case 1: // secure line (until next newline) all tags and commands in MXP are allowed within the line.  When a newline is received from the MUD, the mode reverts back to the Default mode.
+        mMXP_MODE = MXP_MODE_SECURE;
+        break;
+    case 2: // locked line (until next newline) no MXP or HTML commands are allowed in the line.  The line is not parsed for any tags at all.  This is useful for "verbatim" text output from the MUD.  When a newline is received from the MUD, the mode reverts back to the Default mode.
+        mMXP_MODE = MXP_MODE_LOCKED;
+        break;
+    case 3: //  reset (MXP 0.4 or later) - close all open tags.  Set mode to Open.  Set text color and properties to default.
+        mMxpTagBuilder.reset();
+        mMXP_MODE = mMXP_DEFAULT;
+        break;
+    case 4: // temp secure mode (MXP 0.4 or later) - set secure mode for the next tag only.  Must be immediately followed by a < character to start a tag.  Remember to set secure mode when closing the tag also.
+        mMXP_MODE = MXP_MODE_TEMP_SECURE;
+        break;
+    case 5: // lock open mode (MXP 0.4 or later) - set open mode.  Mode remains in effect until changed.  OPEN mode becomes the new default mode.
+        mMXP_DEFAULT = mMXP_MODE = MXP_MODE_OPEN;
```

Comment:
> Whilst multiple assignments like this is perfectly valid code, I'd personally avoid it to prevent confusion with something like: `result = value1 == value2;` - mind you I think rain is wet so who am I to say.

I would avoid it too, but I didn't touch this code, it was exactly like this in TBuffer, I avoided change as much as possible. I didn't even change these variable names, though I don't Like them either.